### DISCUSSION TITLE
Disable push animation for NavigationStackStore

### DISCUSSION THREAD

**Discussion by Otbivnoe** [pointfreeco/swift-composable-architecture#2504] [Q&A]
Hey! Is it possible to disable a push animation for NavigationStackStore? Is there a way of invoking `withAnimation(nil)` directly in a reducer where I mutate a path variable?

**Comment by mbrandonw** [comment]
Hi @Otbivnoe, typically you send actions with animations by doing:

```swift
viewStore.send(.buttonTapped, animation: .default)
```

Whatever state changes are made from sending that action will be done so inside a `withAnimation { … }` block.

However, I'm not sure SwiftUI animations control the push animations for `NavigationStack`. Have you tried this in vanilla SwiftUI to see if it works?

**Reply by Otbivnoe** [reply]
Hey @mbrandonw, thanks for the answer. Seems like it became available in [iOS 16.2](https://developer.apple.com/documentation/ios-ipados-release-notes/ios-ipados-16_2-release-notes#New-Features), but I'm now sure how to achieve it in TCA. Do you have any ideas? 🙏
